packages/cube/lib/batch.js

```diff
@@ -0,0 +1,30 @@
+import { Duplex } from 'stream'
+import rdf from '@zazuko/env-node'
+
+// Iterable<X> => Iterable<X[]>
+export async function * chunkObjectsBySize(size, iterable) {
+  let chunk = []
+  for await (const item of iterable) {
+    chunk.push(item)
+    if (chunk.length === size) {
+      yield chunk
+      chunk = []
+    }
+  }
+  if (chunk.length > 0) {
+    yield chunk
+  }
+}
+
+// Iterable<Dataset> => Iterable<Dataset>
+export async function * chunkBySize(size, iterable) {
+  for await (const array of chunkObjectsBySize(size, iterable)) {
+    const batch = rdf.dataset()
+    for (const dataset of array) {
+      batch.addAll(dataset)
+    }
+    yield batch
+  }
+}
+
+export const batch = size => Duplex.from(iterable => chunkBySize(Number(size), iterable))
```

Comment:
Is `chunkBySize` ever imported anywhere? I ask because you'd want to pass RDF env from the step factory

```suggestion
export async function * chunkBySize({ env }, size, iterable) {
  for await (const array of chunkObjectsBySize(size, iterable)) {
    const batch = env.dataset()
    for (const dataset of array) {
      batch.addAll(dataset)
    }
    yield batch
  }
}

export const batch = function (size) => {
  return Duplex.from(iterable => chunkBySize(this, Number(size), iterable))
}
```

This is an example of how you'd use the env from barnard context

Reply:
not used elsewhere, we can remove the `export` and use the env from barnard context